store/rootmulti/store.go

```diff
@@ -402,19 +414,17 @@ func (rs *Store) Commit() types.CommitID {
 		version = previousHeight + 1
 	}
 
-	rs.lastCommitInfo = rs.commitStores(version, rs.stores)
-
-	var pruneErr error
-	defer func() {
-		rs.flushMetadata(rs.db, version, rs.lastCommitInfo)
-		if pruneErr != nil {
-			panic(pruneErr)
-		}
-	}()
+	newCommitInfo := rs.commitStores(version, rs.stores)
+	rs.updateLatestCommitInfo(newCommitInfo, version)
 
-	pruneErr = rs.handlePruning(version)
+	err := rs.handlePruning(version)
+	if err != nil {
+		panic(err)
+	}
 
+	rs.mx.RLock()
```

Comment:
This might be redundant because we only write in the same goroutine above. However, there can be concurrent readers.

What do reviewers think?